model/atmosphere/diffusion/src/icon4py/model/atmosphere/diffusion/diffusion.py

```diff
@@ -397,7 +397,7 @@ def __init__(
         self._determine_horizontal_domains()
 
         self.mo_intp_rbf_rbf_vec_interpol_vertex = setup_program(
-            backend=self._backend,
+            backend=backend,
```

Comment:
after this change, do we need to store `backend` in the instance or can we drop it?

Reply:
I've reverted back to the original version as the backend argument is also used in `_allocate_local_fields`